### PR TITLE
AEA-1439 Remove requirement for exact protocol buffer compiler version

### LINKED ISSUE #2148: Lift requirement to use a specific libprotoc version
**Is your feature request related to a problem? Please describe.**
Currently, we force usage of `libprotoc` version `3.12.4`.

**Describe the solution you'd like**
Lift the constraint to use that specific version. This can be done once the protobuf files are generated during the build stage.

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR:
- Remove requirement for exact libprotoc version when generating protocols.
- Add aea version and libtprotoc version tag in protocol package init file

## Fixes

#2148 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: aea/protocols/generator/common.py [hunk 1]
```diff
@@ -364,14 +364,13 @@ def check_prerequisites() -> None:
...
+def get_protoc_version() -> str:
+    """Get the protoc version used."""
     result = subprocess.run(  # nosec
         ["protoc", "--version"], stdout=subprocess.PIPE, check=True
     )
     result_str = result.stdout.decode("utf-8")
-    if LIBPROTOC_VERSION not in result_str:
-        raise FileNotFoundError(  # pragma: nocover
-            f"Invalid version for protoc. Found: {result_str}. Required: {LIBPROTOC_VERSION}."
-        )
```
Comment (DavidMinarsch) [inline aea/protocols/generator/common.py:374]: We could keep a minimum version check, but I think its not needed as the protobuf schema will enforce that.

File: packages/fetchai/protocols/contract_api/__init__.py [hunk 2]
```diff
@@ -17,7 +17,12 @@
 #
 # ------------------------------------------------------------------------------
 
-"""This module contains the support resources for the contract_api protocol."""
+"""
+This module contains the support resources for the contract_api protocol.
+
+It was created with protocol buffer compiler version `libprotoc 3.11.4
+` and aea version `1.0.0rc1`.
```
Comment (DavidMinarsch) [inline packages/fetchai/protocols/contract_api/__init__.py:24]: The reason why we kept the exact version was to be able to recreate the exact protocol. By committing this information here a developer can recreate the exact protocol folder when using the relevant versions.

Comment (marcofavorito) [inline packages/fetchai/protocols/contract_api/__init__.py:24]: the backtick '`' in a new line is a bit unfortunate; consider removing them

**Review by marcofavorito — APPROVED** [review]
LGTM

File: aea/protocols/generator/common.py [hunk 3]
```diff
@@ -364,14 +364,13 @@ def check_prerequisites() -> None:
             "Cannot find protocol buffer compiler! To install, please follow this link: https://developers.google.com/protocol-buffers/"
         )
 
+def get_protoc_version() -> str:
+    """Get the protoc version used."""
     result = subprocess.run(  # nosec
         ["protoc", "--version"], stdout=subprocess.PIPE, check=True
     )
     result_str = result.stdout.decode("utf-8")
```
Comment (marcofavorito) [inline aea/protocols/generator/common.py:372]: then why we need this?

File: packages/fetchai/protocols/contract_api/__init__.py (hunk 2 above)
Comment (marcofavorito) [reply packages/fetchai/protocols/contract_api/__init__.py:24]: or put each dependency version in a new line, e.g.
```
libprotoc version: ...
aea version: ...
```

**Review by 5A11 — APPROVED** [review]
LGTM

**Review by solarw — APPROVED** [review]
LGTM

**Pull request merged**